error/s2n_errno.h

```diff
@@ -312,6 +312,7 @@ typedef enum {
     S2N_ERR_KTLS_UNSUPPORTED_CONN,
     S2N_ERR_KTLS_ULP,
     S2N_ERR_KTLS_ENABLE_CRYPTO,
+    S2N_ERR_KTLS_BAD_CMSG,
     S2N_ERR_ATOMIC,
```

Comment:
This definitely isn't a usage error. Probably an IO error.

Reply:
Can I make a new section with ktls specific errors? That seems nicer. Or if we think that we will reuse this error then we should remove the KTLS prefix. I kind of like that since we could theoritically use sendmsg and recvmsg later on.